anaconda_anon_usage/patch.py

```diff
@@ -1,96 +1,42 @@
-    client_token = ""
-    _debug("%s token path: %s", what.capitalize(), fpath)
-    if exists(fpath):
-        try:
-            # Use just the first line of the file, if it exists
-            client_token = "".join(open(fpath).read().splitlines()[:1])
-            _debug("Retrieved %s token: %s", what, client_token)
-        except Exception as exc:
-            _debug("Unexpected error reading: %s\n  %s", fpath, exc, error=True)
-    if len(client_token) < 22:
-        if len(client_token) > 0:
-            _debug("Generating longer token")
-        client_token = get_random_token()
-        try:
-            os.makedirs(dirname(fpath), exist_ok=True)
-            with open(fpath, "w") as fp:
-                fp.write(client_token)
-            _debug("Generated new token: %s", client_token)
-            _debug("%s token saved: %s", what.capitalize(), fpath)
-        except Exception as exc:
-            _debug("Unexpected error writing: %s\n  %s", fpath, exc, error=True)
-            client_token = ""
-    return client_token
-
-
-def get_client_token():
-    fpath = join(expanduser("~/.conda"), "aau_token")
-    return get_saved_token(fpath, "client")
-
-
-def get_environment_token():
-    try:
-        prefix = Context.checked_prefix or context.target_prefix
-        if prefix is None:
-            return None
-    except Exception as exc:
-        _debug("error retrieving prefix: %s", exc)
-        return None
-    fpath = join(prefix, "etc", "aau_token")
-    return get_saved_token(fpath, "environment")
-
-
-@functools.lru_cache(maxsize=None)
-def client_token_string():
-    parts = ["aau/" + __version__]
-    value = get_client_token()
-    if value:
-        parts.append("c/" + value)
-    parts.append("s/" + get_random_token())
-    value = get_environment_token()
-    if value:
-        parts.append("e/" + value)
-    result = " ".join(parts)
-    _debug("Full client token: %s", result)
-    return result
+from .api import _debug, token_string
 
 
 def _new_user_agent(ctx):
     result = ctx._old_user_agent
     if context.anaconda_anon_usage:
-        token = client_token_string()
-        if token:
-            result += " " + token
+        prefix = Context.checked_prefix or context.target_prefix or sys.prefix
+        if prefix:
+            token = token_string(prefix)
+            if token:
+                result += " " + token
     else:
         _debug("anaconda_anon_usage disabled by config")
     return result
 
 
+def _new_check_prefix(prefix, json=False):
+    Context.checked_prefix = prefix
+    Context._old_check_prefix(prefix, json)
+
+
+def _patch_check_prefix():
+    _debug("Applying anaconda_anon_usage cli.install patch")
+    from conda.cli import install as cli_install
+
+    Context._old_check_prefix = cli_install.check_prefix
+    cli_install.check_prefix = _new_check_prefix
+    context._initialized = True
+
+
 def main(plugin=False):
     if hasattr(Context, "_old_user_agent"):
```

Comment:
Just double-checking, can't we use that now?

```suggestion
    if hasattr(context, "_aau_initialized"):
```